### PR TITLE
[api-explorer] Convert flow to Typescript

### PR DESCRIPTION
**Pull request by imoans**
https://github.com/phenyl-js/phenyl/issues/164

### CODE REVIEW

File: modules/api-explorer/package.json [hunk 1]
```diff
@@ -1,28 +1,32 @@
...
+  "main": "lib/PhenylApiExplorer.js",
+  "types": "lib/PhenylApiExplorer.d.js",
+  "repository": "https://github.com/phenyl-js/phenyl/tree/master/modules/phenyl-api-explorer",
   "scripts": {
+    "build": "upbin tsc --declaration",
+    "type-check": "upbin tsc --noEmit",
     "start": "nodemon examples/index.js --exec babel-node --watch src --watch examples",
     "test": "echo no test specified in phenyl-api-explorer"
   },
   "author": "Leko <leko.noor@gmail.com>",
```
Comment (Naturalclar) [inline modules/api-explorer/package.json]: You can add your own name in the author if you wish :)

File: modules/api-explorer/examples/index.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
+import http from "http";
+import PhenylRestApi from "phenyl-rest-api";
```
Comment (Naturalclar) [inline modules/api-explorer/examples/index.ts]: should be changed to `@phenyl/rest-api`

File: modules/api-explorer/examples/index.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+import { createEntityClient } from "phenyl-memory-db";
```
Comment (Naturalclar) [inline modules/api-explorer/examples/index.ts]: could you add a `TODO:` comment to change this to `@phenyl/memory-db` once memory-db is implemented?

File: modules/api-explorer/examples/index.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+import {
+  StandardUserDefinition,
+  StandardEntityDefinition
+} from "phenyl-standards";
+import PhenylHttpServer from "phenyl-http-server";
```
Comment (Naturalclar) [inline modules/api-explorer/examples/index.ts]: could you add a TODO comment to change this to `@phenyl/http-server`?

File: modules/api-explorer/examples/index.ts [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+const PORT = 8000;
+const memoryClient = createEntityClient();
+
+class HospitalDefinition extends StandardEntityDefinition {
+  async authorization(
+    reqData: GeneralRequestData,
+    session: Session | null
+  ): Promise<boolean> {
+    // eslint-disable-line no-unused-vars
```
Comment (Naturalclar) [inline modules/api-explorer/examples/index.ts]: This `eslint-disable` is not disabling anything. Could you delete the comment?

File: modules/api-explorer/package.json [hunk 6]
```diff
@@ -1,28 +1,32 @@
 {
-  "name": "phenyl-api-explorer",
-  "version": "0.1.16",
+  "name": "@phenyl/api-explorer",
+  "version": "1.0.0",
   "description": "Automatically renders API explorer pages like swagger",
-  "main": "./dist/PhenylApiExplorer.js",
-  "jsnext:main": "./jsnext.js",
+  "main": "lib/PhenylApiExplorer.js",
+  "types": "lib/PhenylApiExplorer.d.js",
```
Comment (Naturalclar) [inline modules/api-explorer/package.json]: types should be `lib/PhenylApiExplorer.d.ts`

File: modules/api-explorer/examples/index.ts (hunk 3 above)
Comment (Naturalclar) [reply modules/api-explorer/examples/index.ts]: memory-db has been converted to TypeScript. Please use `@phenyl/memory-db` instead

File: modules/api-explorer/examples/index.ts (hunk 4 above)
Comment (Naturalclar) [reply modules/api-explorer/examples/index.ts]: http-server has been converted to TypeScript. please use `@phenyl/http-server`

**Comment by imoans** [comment]
@Naturalclar 
I fixed the problems. Could you review again?

File: modules/api-explorer/package.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  "author": [
+    "Leko <leko.noor@gmail.com>",
+    "imoans <naomins0101@gmail.com>"
+  ],
+  "license": "Apache-2.0",
+  "devDependencies": {
+    "@phenyl/interfaces": "~1.0.0",
+    "@types/ejs": "^2.6.2",
+    "nodemon": "^1.18.3",
+    "phenyl-http-client": "~0.2.13",
```
Comment (Naturalclar) [inline modules/api-explorer/package.json]: these imports should be changed to `@phenyl/xxx : "~1.0.0"`

File: modules/api-explorer/package.json [hunk 8]
```diff
@@ -25,7 +25,7 @@
     "@phenyl/lambda-adapter": "~1.0.0",
     "@phenyl/memory-db": "~1.0.0",
     "@phenyl/rest-api": "~1.0.0",
-    "@phenyl/standards": "~1.0.0"
+    "phenyl-standards": "~0.2.9"
```
Comment (Naturalclar) [inline modules/api-explorer/package.json]: phenyl-standards should be changed to `@phenyl/standards` now. please rebase from the latest typescript branch

File: modules/api-explorer/examples/index.ts [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+const functionalGroup: FunctionalGroup = {
+  customQueries: {
+    test: new TestCustomCommand()
+  },
+  customCommands: {
+    test: new TestCustomQuery()
+  },
+  users: {
+    // @ts-ignore TODO Convert phenyl-standards to TS
```
Comment (Naturalclar) [inline modules/api-explorer/examples/index.ts]: could you remove these ts-ignore and check if type-check passes?

File: modules/api-explorer/src/PhenylApiExplorer.ts [hunk 10]
```diff
@@ -0,0 +1,69 @@
+/* eslint-env node */
```
Comment (Naturalclar) [inline modules/api-explorer/src/PhenylApiExplorer.ts]: this comment can be deleted

**Pull request merged**